### PR TITLE
Move search suggestion logic to selector

### PR DESCRIPTION
**Pull request by seanyesmunt**
### Changes
- Default first suggestion to be a search
- Most of the search suggestion logic is now in a selector
- Just pass in the api results to the search reducer with a `query` key so previous suggestions are preserved when using the backspace/retyping the same query

### CODE REVIEW

File: src/redux/selectors/search.js [hunk 1]
```diff
@@ -37,3 +41,79 @@ export const selectWunderBarAddress = createSelector(
 );
 
 export const selectSearchBarFocused = createSelector(selectState, state => state.focused);
+// export const selectSear
+
+export const selectSearchSuggestions = createSelector(
```
Comment (akinwale) [inline src/redux/selectors/search.js]: Just an idea I had. I was thinking we should have an additional selector called `selectLocalSearchSuggestions` which could possibly be used in cases where an API call doesn't necessarily have to be made.

Comment (seanyesmunt) [reply src/redux/selectors/search.js]: Maybe based on previous searches?

**Comment by seanyesmunt** [comment]
@akinwale I'll create a PR to lbry-android after adding this into the app. The changes should be very minor.

File: src/redux/selectors/search.js (hunk 1 above)
Comment (akinwale) [reply src/redux/selectors/search.js]: Yeah, that makes sense.

**Pull request merged**